File: leetcode/0121.best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        l,r = 0,1
        profit = 0
        while( r < len(prices)):
            if prices[l] < prices[r]:
                diff = prices[r] - prices[l]
                profit = max(profit, diff)
            else:
                l = r
            r += 1
        return profit
        

File: leetcode/0121.best-time-to-buy-and-sell-stock.ts
function maxProfit(prices: number[]): number {
    let l = 0;
    let r = 1;
    let N = prices.length - 1;
    let profit = 0;
    while(r <= N) {
        let diff = prices[r] - prices[l];
        profit = Math.max(profit, diff);
        while(prices[l] > prices[r]) {
            l = l + 1
        }
        r = r + 1 
    }
    return profit
};